diff --git a/subprojects/nanobind-2.4.0/meson.build b/subprojects/nanobind-2.4.0/meson.build
index 97c336f..ff40803 100644
--- a/meson.build
+++ b/meson.build
@@ -42,7 +42,13 @@ endif
 
 compiler = meson.get_compiler('cpp')
 
-add_project_arguments(compiler.get_supported_arguments('/D_CRT_SECURE_NO_WARNINGS'), language: 'cpp')
+add_project_arguments(compiler.get_supported_arguments(
+  '/D_CRT_SECURE_NO_WARNINGS',
+  '-Wno-zero-length-array',
+  '-Wno-nested-anon-types',
+  '-Wno-pedantic',
+  '-Wno-sign-compare'
+), language: 'cpp')
 
 cxx_id = compiler.get_id()
 # Discussion in WrapDB PR #1556 as to whether or not this is necessary
